static/app/components/dropdownLink.spec.tsx

```diff
@@ -0,0 +1,314 @@
+        render(
+          <DropdownLink alwaysRenderMenu={false} title="test">
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        expect(screen.queryByText('hi')).not.toBeInTheDocument();
+
+        // open
+        userEvent.click(screen.getByText('test'));
+
+        expect(screen.getByText('hi')).toBeInTheDocument();
+      });
+    });
+
+    describe('While Opened', function () {
+      it('closes when clicked outside', async function () {
+        jest.useFakeTimers();
+
+        render(
+          <div data-test-id="outside-element">
+            <DropdownLink title="test" alwaysRenderMenu={false}>
+              <li>hi</li>
+            </DropdownLink>
+          </div>
+        );
+
+        // Open menu
+        userEvent.click(screen.getByText('test'));
+
+        // Click outside
+        userEvent.click(screen.getByTestId('outside-element'));
+
+        jest.runAllTimers();
+
+        await waitForElementToBeRemoved(() => screen.getByText('hi'));
+      });
+
+      it('closes when dropdown actor button is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false}>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Open menu
+        userEvent.click(screen.getByText('test'));
+
+        // Click again
+        userEvent.click(screen.getByText('test'));
+
+        expect(screen.queryByText('hi')).not.toBeInTheDocument();
+      });
+
+      it('closes when dropdown menu item is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false}>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Open menu
+        userEvent.click(screen.getByText('test'));
+
+        userEvent.click(screen.getByText('hi'));
+
+        expect(screen.queryByText('hi')).not.toBeInTheDocument();
+      });
+
+      it('does not close when menu is clicked and `keepMenuOpen` is on', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false} keepMenuOpen>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Open menu
+        userEvent.click(screen.getByText('test'));
+
+        // Click again
+        userEvent.click(screen.getByText('test'));
+
+        expect(screen.getByText('test')).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Controlled', function () {
+    describe('Opened', function () {
+      it('does not close when menu is clicked', function () {
+        render(
+          <DropdownLink title="test" alwaysRenderMenu={false} isOpen>
+            <li>hi</li>
+          </DropdownLink>
+        );
+
+        // Click option
+        userEvent.click(screen.getByText('hi'));
+
+        // Should still be open
+        expect(screen.getByText('hi')).toBeInTheDocument();
+      });
+
+      it('does not close when document is clicked', function () {
+        jest.useFakeTimers();
```

Comment:
would it be fine if instead of using `jest.useFakeTimers` and `jest.runAllTimers()`, we would just wait for the element by using `await screen.find*`?

Reply:
We still want to use `jest.useFakeTimers()` so the test can finish as fast as possible, but I just learned that the async utilities do advance fake timers which is great: https://github.com/testing-library/dom-testing-library/issues/661

So I kept the fake timers and removed all the `runAllTimers()` calls